system/jlib/jfile.cpp

```diff
@@ -760,6 +760,69 @@ bool CFile::remove()
 #endif
 }
 
+static std::atomic<unsigned> renameRetries{(unsigned)-1}; // initially uninitialized, set on 1st use (in rename)
+
+void setRenameRetries(unsigned numRetries)
+{
+    renameRetries = numRetries;
+}
+
+static void doRename(const char *src, const char *dst, const char *errMsg)
+{
+    unsigned retriesLeft = renameRetries;
+    unsigned retry = 0;
+    while (true)
+    {
+        if ((unsigned)-1 == retriesLeft)
+        {
+            // NB: potentially could be >1 thread here, but that's ok.
+            try
+            {
+                retriesLeft = getGlobalConfigSP()->getPropInt("misc/@numRenameRetries");
+            }
+            catch (IException *e) // handle cases where config. not available
+            {
+                EXCLOG(e, "doRename");
+                e->Release();
+                retriesLeft = 0;
+            }
+            dbgassertex((unsigned)-1 != retriesLeft);
+            setRenameRetries(retriesLeft);
+        }
+        if (-1 != ::rename(src, dst))
+            return; // success
+
+        if (0 == retriesLeft)
```

Comment:
Would it be useful to add the tracing even if retries are 0, e.g. for components that haven't changed the option.

Reply:
could do. I deliberatly kept it to default 0 and no tracing etc. in normal case, because I was slightly concerned there was exisitng code that might be using rename in a try and see type manner.

That was probably being overly cautious.

I can change it so all components trace.